internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go

```diff
@@ -813,6 +815,9 @@ func (h *Handler) listSharesWithMe(w http.ResponseWriter, r *http.Request) {
 			continue
 		}
 		h.addDisplaynames(r.Context(), gwc, data)
+		// Needed because received shares can be jailed in a folder in the users home
+		data.FileTarget = path.Join(h.sharePrefix, path.Base(info.Path))
+		data.Path = path.Join(h.sharePrefix, path.Base(info.Path))
```

Comment:
hm, the info.Path should already contain the share prefix ... ah wait, the request firts queries the share manager, which only knows file ids, and then sends a stat request to the storage provider to fill in the file details.

@labkode Another reason whd I think we need a service that keeps track of mountpoints. IMO that is the storage registry. It should know the file reference and the mount point. then it can cache the etag for every root, especially the shares. We can then change this hardcoded SharePrefix to an arbitrary location in the users home and calculate the etag based on the information that is cached in the storage registry.

For now this is LGTM

Reply:
> hm, the info.Path should already contain the share prefix ... ah wait, the request firts queries the share manager, which only knows file ids, and then sends a stat request to the storage provider to fill in the file details.

Yes, exactly that was the problem here. :see_no_evil: